server/data/restClientBuilder.ts

```diff
@@ -0,0 +1,137 @@
+import superagent from 'superagent'
+import Agent, { HttpsAgent } from 'agentkeepalive'
+import logger from '../../log'
+import { buildErrorHandler } from './clientErrorHandler'
+
+interface TimeoutSpec {
+  response: number
+  deadline: number
+}
+
+interface AgentConfig {
+  maxSockets: number
+  maxFreeSockets: number
+  freeSocketTimeout: number
+}
+
+export interface RestClientConfig {
+  timeout: TimeoutSpec
+  agent: AgentConfig
+}
+
+const NOT_FOUND = 404
+const UNAUTHORIZED = 401
+
+interface TokenSource {
+  (): Promise<string>
+}
+
+export const constantTokenSource = (token): TokenSource => async () => {
+  if (!token) {
+    throw Error('Unauthorised access')
+  }
+  return token
+}
+
+export const dynamicTokenSource = (signInService, oauthServiceName: string): TokenSource => async () => {
```

Comment:
could possibly call this something about client credentials / anonymous / system to system token?

Reply:
Done